viewing_party/party.py

```diff
@@ -1,23 +1,320 @@
+
+    for movie in user_watched:
+        if movie not in friends_watched:
+            unique_watched.append(movie)
+
+    return unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    """
+    This function determines which movies the user has not watched 
+    but at least one of the friends have watched.
+    
+    Parameters:
+        user_data(dict): has two keys:
+            "watched": List of movie dictionaries the user has watched, each with a "title".
+            "friends": List of friends, each having a "watched" list of movie dictionaries.
+
+    Returns:
+        list of dict of unique movies.
+    """
+    
+    user_watched = user_data["watched"]
+    friends_watched = get_friends_watched(user_data)
+
+    unique_friends_watched = []
+
+    for movie in friends_watched:
+        if movie not in user_watched and movie not in unique_friends_watched:
+            unique_friends_watched.append(movie)
+    
+    return unique_friends_watched
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs (user_data):
+    """
+    Return a list of recommended movies based on the user's subscriptions 
+        and their's friends watched movies
+        
+    Parameters:
+        user_data (dict): has two keys:
+            "subscriptions": List of streaming services the user has access to
+            "friends": List of friends, each having a "watched" list of movie dictionaries, 
+                each movie has a "host"
+    
+    Returns:
+        list of dict of movies
+    """
+    
+    user_watched = user_data["watched"]
+    friends_watched = get_friends_watched(user_data)
+  
+    recommended_movies = []
+    for movie in friends_watched:
+        if movie not in user_watched and movie["host"] in user_data["subscriptions"] \
+            and movie not in recommended_movies:
+            recommended_movies.append(movie)
+    
+    return recommended_movies   
+           
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    """
+    Determine a list of recommended movies
+    
+    Parameters:
+        user_data (dict)
+    Notes:
+        user has not watched that movie
+        at leats one of user's friends has watched it
+        "genre" == user's frequent genre
+    Returns:
+        recomemded movies (list)
+    """
+    
+    user_watched = user_data["watched"]
+    friends_watched = get_friends_watched(user_data)
+    most_frequenty_genre = get_most_watched_genre(user_data)
+    
+    # Returns a list of recommended movies based in user's most frequently watched genre.
+    recommended_movies = []
+    for movie in friends_watched:
+        if movie not in user_watched and \
+            movie["genre"] == most_frequenty_genre and\
+            movie not in recommended_movies:

```

Comment:
Same idea here as before. Every other collection creation should check for duplicates, which means this conditional should be unnecessary!